### PR TITLE
Reflect text changes and expose margin settings

### PR DESCRIPTION
**Pull request by svanherk**
There are two things we need in order to make the UI filter component, both in this PR (you can look at the commits separately for each change):
- If the d2l-tab-panel's text changes, that change will be propagated through to the tab component
- We want to be able to control the left margin, right now it looks like this:
image
And we'd like to be able to set it to this:
image

### CODE REVIEW

File: d2l-tab.js [hunk 1]
```diff
@@ -38,12 +38,12 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-tab">
 			}
 
 			:host(:first-child) .d2l-tab-text {
-				margin-left: 0;
+				margin-left: var(--d2l-tab-first-child-margin, 0);
```
Comment (dbatiste) [inline d2l-tab.js]: Is white-space needed on the right side as well?  i.e. is there a scenario where the tab text would be up against the right side of the dropdown?  I am wondering if a better approach is for the container to apply padding?

File: d2l-tab.js [hunk 2]
```diff
@@ -60,13 +60,13 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-tab">
 			}
 
 			:host(:first-child) .d2l-tab-selected-indicator {
-				margin-left: 0;
-				width: calc(100% - 0.6rem);
+				margin-left: var(--d2l-tab-first-child-margin, 0);;
+				width: calc(100% - 0.6rem - var(--d2l-tab-first-child-margin, 0rem));
```
Comment (dbatiste) [inline d2l-tab.js]: Nit: units on 0 value.

Comment (svanherk) [reply d2l-tab.js]: So I had this as just `0`, and the whole thing complete broke until I added units.  In Chrome!

File: d2l-tabs.js [hunk 3]
```diff
@@ -715,6 +718,15 @@ Polymer({
 
 	},
 
+	_handleTabTextUpdate: function(e) {
+		var panel = dom(e).rootTarget;
+		var tab = dom(this.root).querySelector('[aria-controls="' + panel.id + '"]');
```
Comment (dlockhart) [inline d2l-tabs.js]:
```suggestion
		var tab = dom(this.root).querySelector(`[aria-controls='${panel.id}']`);
```

**Comment by dlockhart** [comment]
For the margin change: what do we think about treating it as an attribute/mode instead of allowing people to specify the exact margins? It would only work if design agreed that they'd never want a different margin. For example (probably better name for the attribute): `<d2l-tabs fit-mode="tight|spaced">`. The `tight` fit mode would be default and is how it behaves today, whereas the `spaced` (need a better name) fit mode would be for when tabs are inside a container like this and need some extra spacing.

**Comment by svanherk** [comment]
Hmm I do like that idea, since the only use cases I see right now are "have it match the margin on the right side" or "left-align it" (the only current option).  Jeff G was the designer for this, correct?

**Comment by dbatiste** [comment]
I think we should get Jeff's opinion, but text should probably be lined up.  A while back (when I first saw this pattern before My Courses) I thought there might be some use for a `d2l-dropdown-tabs` content component that encapsulates/standardizes this layout detail for the dropdown specific case.  I think I also saw this pattern proposed for the user-settings dropdown in nav.  Not sure it it's needed, but if there was focus behavior is would become more compelling.  Scrolling of long content is another use-case that might come up - i.e. how does that look when there are tabs at the top?

dropdown-tabs-content

**Comment by svanherk** [comment]
We've been talking about the whitespace changes in other threads, so to summarize here:
- We need some more design about how this should work (should the consumer control white space, or should they be given two options (tight or contained)?)
- I can get this working for our use-case that doesn't have a solid design around it, so I'm going to defer this change for now (that commit has been removed)
- When we circle back to this, we may want some type of `d2l-dropdown-tabs-content` component to handle the margins and setup for the next person wanting to use tabs within a dropdown

File: d2l-tabs.js [hunk 4]
```diff
@@ -715,6 +718,15 @@ Polymer({
 
 	},
 
+	_handleTabTextUpdate: function(e) {
```
Comment (dbatiste) [inline d2l-tabs.js]: Do we need to add a `.bind(this)` in the `ready` for this method?

Comment (svanherk) [reply d2l-tabs.js]: It doesn't seem to need it, but TBH I'm not sure why... I can add it anyways

**Pull request merged**